plugins/skipmap/index.js

```diff
@@ -0,0 +1,158 @@
+    let votePos = 0;
+    let voteNeg = 0;
+    let intervalReminderBroadcasts;
+    let timeoutVote;
+    let timeLastVote;
+    let voteActive;
+    const playerVotes = {};
+
+    server.on(CHAT_MESSAGE, (info) => {
+      // Run through conditions
+      // check if message is command
+      if (!info.message.chat.startsWith(options.command)) return;
+      // check if enough time has passed since start of round and if not, inform the player
+      if (server.layerHistory[0].time < options.startTimer) {
+        const min = Math.floor(
+          ((options.startTimer - server.layerHistory[0].time) % (1000 * 60 * 60)) / (1000 * 60)
+        );
+        const sec = Math.floor(
+          (options.startTimer - server.layerHistory[0].time) % ((1000 * 60) / 1000)
+        );
+        server.rcon.warn(
+          info.steamID,
+          'Not enough time has passed since the start of the match. Please try again in ' +
+            min +
+            'm ' +
+            sec +
+            's'
+        );
+        return;
+      }
+      // check if enough time remains in the round, if not, inform player
+      if (server.layerHistory[0].time < 2 * 60 * 60 * 1000 - options.endTimer) {
+        server.rcon.warn(info.steamID, 'Too close to expected end of match.');
+        return;
+      }
+
+      // check if enough time has passed since the last vote
+      if (timeLastVote - new Date() < options.pastVoteTimer) {
+        server.rcon.warn(info.steamID, 'Not enough time has passed since the last vote.');
+        return;
+      }
+
+      // Actual vote
+      voteActive = true;
+      votePos = 1;
+      playerVotes[info.steamID] = info.message;
+      voteNeg = 0;
+      // Set reminders
+      intervalReminderBroadcasts = setInterval(() => {
+        server.broadcast(
+          'A vote to skip the current map has started. Please vote in favour with + or against with -.'
+        );
+        server.broadcast('Current counter is: ' + votePos + ' in favour, ' + voteNeg + ' against.');
+      }, options.reminderInterval);
+
+      // End vote
+      // Disable recording of new votes, stop further broadcasts
+      timeoutVote = setTimeout(() => {
+        voteActive = false;
+        clearInterval(intervalReminderBroadcasts);
+        // Check if enough people voted
+        if (voteNeg + votePos < options.minVoteCount) {
+          server.rcon.broadcast('Not enough people voted for the vote to go through.');
+          return;
+        }
+        if (votePos > voteNeg) {
+          server.rcon.execute('AdminEndMatch');
+        } else {
+          server.rcon.broadcast(
+            'Not enough people voted in favour of skipping the match. ' +
+              votePos +
+              ' voted in favour, ' +
+              voteNeg +
+              'against.'
+          );
+        }
+      }, options.voteDuration);
+
+      // As a vote happened, stop any further votes from happening until enough time has passed
+      timeLastVote = new Date();
```

Comment:
I'd maybe move this up into the timeout. I think it would be better to have no vote within x minutes of the last one ending as opposed to x minutes of the last on starting.